wear/src/main/java/com/android/gpstest/wear/MainActivity.kt

```diff
@@ -0,0 +1,419 @@
+                userDeniedPermission = false
+                gpsStart()
+            } else {
+                userDeniedPermission = true
+            }
+        }
+    }
+
+    @ExperimentalCoroutinesApi
+    @SuppressLint("MissingPermission")
+    @Synchronized
+    private fun gpsStart() {
+        PreferenceUtils.saveTrackingStarted(true, prefs)
+
+        // Observe flows
+        observeLocationFlow()
+    }
+
+    @ExperimentalCoroutinesApi
+    private fun observeLocationFlow() {
+        // This should be a Flow and not LiveData to ensure that the Flow is active before the Service is bound
+        if (locationFlow?.isActive == true) {
+            // If we're already observing updates, don't register again
+            return
+        }
+        // Observe locations via Flow as they are generated by the repository
+        locationFlow = repository.getLocations()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
+            .onEach {
+                //lastLocation = it
+                Log.d(TAG, "Activity location: $it")
+            }
+            .launchIn(lifecycleScope)
+    }
+
+    @Synchronized
+    private fun gpsStop() {
+        PreferenceUtils.saveTrackingStarted(false, prefs)
+        locationFlow?.cancel()
+    }
+
+    companion object {
+        private const val TAG = "MainActivity"
+    }
+}
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@Composable
+fun WearApp(satStatues: List<String>, signalInfoViewModel: SignalInfoViewModel) {
+    val gnssStatuses: List<SatelliteStatus> by signalInfoViewModel.filteredGnssStatuses.observeAsState(emptyList())
+    val location: Location by signalInfoViewModel.location.observeAsState(Location("default"))
+    val fixState: FixState by signalInfoViewModel.fixState.observeAsState(FixState.NotAcquired)
+    GpstestTheme {
+        val listState = rememberScalingLazyListState()
+        Scaffold(
+            timeText = {
+                if (!listState.isScrollInProgress) {
+                    TimeText(
+                        timeSource = TimeTextDefaults.timeSource(
+                            DateFormat.getBestDateTimePattern(Locale.getDefault(), "hh:mm:ss")
+                        )
+                    )
+                }
+            },
+            positionIndicator = {
+                PositionIndicator(
+                    scalingLazyListState = listState
+                )
+            }
+        ) {
+            val contentModifier = Modifier
+                .fillMaxWidth()
+                .padding(bottom = 8.dp)
+
+            ScalingLazyColumn(
+                modifier = contentModifier,
+                autoCentering = AutoCenteringParams(itemIndex = 0),
+                state = listState
+            ) {
+                item{
+                    CustomLinearProgressBar(fixState)
+                }
+
+                for (satStatue in satStatues) {
```

Comment:
This FOR block needs to be replaced with data from the `location` object, which will be updated on each new GNSS fix.

Reply:
> This FOR block needs to be replaced with data from the `location` object, which will be updated on each new GNSS fix.

Done